src/AMSlib/wf/resource_manager.hpp

```diff
@@ -145,7 +145,7 @@ class ResourceManager
   static void deallocate(std::vector<T*>& dPtr, AMSResourceType resource)
   {
     for (auto* I : dPtr)
-      RMAllocators[resource]->deallocate(I);
+      RMAllocators[resource]->deallocate(I, dev);
```

Comment:
I think over here you need to delete "dev".

Reply:
Done.